### PR TITLE
tools: clippy: list all complexity and style lints

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview

This lists out which complexity and styles lints we currently need to allow. I grouped them based on lints that seem like reasonable issues we should fix, lints we may want to fix depending on the diff, and lints we most likely will not use.

Part of #3572.

This is potentially an important change because we are adding many clippy lints that we weren't previously.

### Testing Strategy

travis

### TODO or Help Wanted

n/a

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

**Review by brghena — APPROVED** [review]

File: tools/run_clippy.sh [hunk 1]
```diff
@@ -37,4 +34,123 @@ CLIPPY_ARGS="
 -D clippy::wildcard_imports
 "
 
-cargo clippy -- $CLIPPY_ARGS
+# Disallow all complexity lints, then re-allow each one Tock does not comply
+# with.
+#
+# There are three sections:
+# 1. The first section are lints we almost certainly don't want.
+# 2. The section section are lints we may not want, we probably have to see the
```
Comment (hudson-ayers) [inline tools/run_clippy.sh:42]:
```suggestion
# 2. The second section are lints we may not want, we probably have to see the
```

File: tools/run_clippy.sh [hunk 2]
```diff
@@ -37,4 +34,123 @@ CLIPPY_ARGS="
...
+#    resulting diff.
+# 3. The third section are lints that we do want we just need to fixup the code
+#    to pass the lint checks.
+CLIPPY_ARGS_COMPLEXITY="
+-D clippy::complexity
```
Comment (hudson-ayers) [inline tools/run_clippy.sh:47]: I might put the complexity link on a separate line, at a glance it looks like you accidentally denied one of the lints from the first section that you meant to allow

**Review by hudson-ayers — APPROVED** [review]
I don't have a ton of thoughts on this particular grouping, but this approach seems good to me. I assume we already pass a lot of these

**Comment by bradjc** [comment]
> I assume we already pass a lot of these

We pass none of these. That is how I found them.

**Review by ppannuto — APPROVED** [review]

**Comment by bradjc** [comment]
Do we want to move forward with this? This is a step towards using clippy more generally. I have a few branches with fixes/changes to meet various clippy lints. Luckily `clippy --fix` is a thing.

**Review by lschuermann — APPROVED** [review]

**Pull request merged**